transpiler/transpiler.go

```diff
@@ -15,6 +15,39 @@
 //
 
 // transpiler convert a subset of Go to Arduino C++ dialect.
+//
+// Use case is to help port Go code on MCUs (small microcontroller) where there
+// is no operating system. A strict C++ subset (without RTTI) is used.
+//
+// ** Do not take this project too seriously **
+//
+// Challenges and transformations:
+//  - function with multiple return values is converted to returning as a
+//    temporary struct
+//  - string constant is converted to const char *
+//  - interface{} is converted to void *
+//  - interface inheritance is figured out at parsing time
+//  - out of bound check for slice and strings
+//  - string indexing is done via byte offset, not runes
+//  - struct are manually zero initialized
+//  - recursive type resolution of imported packages
+//
+// Out of scope:
+//  - channel type
+//  - defer statement
+//  - go statement
+//  - len function
+//  - map type
+//  - range expression
+//  - select statement
+//  - switch statement
+//  - anonymous function
+//  - function pointer
+//  - pointer to member of struct
+//  - unnamed struct embedding
+//  - memory management, all memory allocation is leaked
+//  - dynamic type casting involving RTTI
+//  - using the STL in the generated code
```

Comment:
This description reminds me a lot of Gomez that @fkautz was working on. I'm just /ccing him here so it's on his radar.